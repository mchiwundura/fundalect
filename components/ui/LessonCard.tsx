import {
  StyleSheet,
  TouchableOpacity,
  useColorScheme,
  View,
} from "react-native";
import { ThemedText } from "../ThemedText";
import { ThemedView } from "../ThemedViewOld";
import { useRouter } from "expo-router";
import { LessonCardProps } from "@/types/ui/LessonCard";

export default function LessonCard(props: LessonCardProps) {
  const router = useRouter();
  const colorScheme = useColorScheme();

  function buttonPressed() {
    props.onPress();
    router.push(props.link);
  }

  return (
    <TouchableOpacity onPress={buttonPressed} style={styles.link}>
      <View style={[styles.background, { backgroundColor: props.color }]} />
      <View
        style={[
          styles.container,
          {
            borderColor: props.color,
            boxShadow:
              colorScheme === "light"
                ? "0 4px 4px rgba(0, 0, 0, 0.25)"
                : "0 4px 4px rgba(0, 0, 0, 0.9)",
          },
        ]}
      >
        <ThemedView
          style={[
            styles.progress,
            { backgroundColor: props.color, width: props.completion * 3.18 },
          ]}
        />
        <ThemedText
          type="defaultSemiBold"
          numberOfLines={1}
          ellipsizeMode="tail"
          style={styles.title}
        >
          {props.title}
        </ThemedText>
      </View>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  link: {
    marginVertical: 10,
    height: 60,
    position: "relative",
  },
  container: {
    padding: 10,
    borderRadius: 8,
    position: "relative",
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    width: "100%",
    flex: 1,
    borderWidth: 3,
    height: 60,
  },
  progress: {
    position: "absolute",
    left: 2,
    top: 2,
    height: 50,
    borderRadius: 5,
  },
  background: {
    position: "absolute",
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    borderRadius: 8,
    opacity: 0.2,
  },
  title: {
    flex: 1,
    marginLeft: 8,
  },
});
