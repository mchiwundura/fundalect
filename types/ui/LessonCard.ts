import { RelativePathString } from "expo-router";

export interface LessonCardProps {
    title: string;
    onPress: any;
    color: string;
    link: RelativePathString;
    completion: number;
}