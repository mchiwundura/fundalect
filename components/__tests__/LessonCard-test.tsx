import { render } from "@testing-library/react-native";
import LessonCard from "../ui/LessonCard";
import { RelativePathString } from "expo-router";

describe("Lesson Card snapshot tests", () => {
    it("renders correctly and matches snapshot", () => {
        const tree = render(
            <LessonCard
                title="Test Lesson"
                onPress={() => {}}
                color="#FF5733"
                link={"/course/1/lesson/1" as RelativePathString}
                completion={0.5}
            />
        );
        expect(tree.toJSON()).toMatchSnapshot();
    });
})