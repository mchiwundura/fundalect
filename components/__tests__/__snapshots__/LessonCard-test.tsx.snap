// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lesson Card snapshot tests renders correctly and matches snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "height": 60,
      "marginVertical": 10,
      "opacity": 1,
      "position": "relative",
    }
  }
>
  <View
    style={
      [
        {
          "borderRadius": 8,
          "bottom": 0,
          "left": 0,
          "opacity": 0.2,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "backgroundColor": "#FF5733",
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 8,
          "borderWidth": 3,
          "flex": 1,
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "space-between",
          "padding": 10,
          "position": "relative",
          "width": "100%",
        },
        {
          "borderColor": "#FF5733",
          "boxShadow": "0 4px 4px rgba(0, 0, 0, 0.25)",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "#fff",
          },
          [
            {
              "borderRadius": 5,
              "height": 50,
              "left": 2,
              "position": "absolute",
              "top": 2,
            },
            {
              "backgroundColor": "#FF5733",
              "width": 1.59,
            },
          ],
        ]
      }
    />
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        [
          {
            "color": "rgba(0,0,0,0.8)",
          },
          undefined,
          undefined,
          {
            "fontSize": 16,
            "fontWeight": "600",
            "lineHeight": 24,
          },
          undefined,
          undefined,
          undefined,
          {
            "flex": 1,
            "marginLeft": 8,
          },
        ]
      }
    >
      Test Lesson
    </Text>
  </View>
</View>
`;
